packages/block-editor/package.json

```diff
@@ -44,6 +44,7 @@
 		"classnames": "^2.2.5",
 		"dom-scroll-into-view": "^1.2.1",
 		"lodash": "^4.17.10",
+		"react-spring": "^8.0.19",
```

Comment:
This module is 47kb minified before gzip, and we're now duplicating it across two of our packages.

At what point should we consider extracting it as a common vendor dependency?

Reply:
I don't know :) What do you think. I don't mind extracting it but it feels very risky in terms of BC as I expect breaking changes in this kind of modules way more often than React for instance.

Reply:
> I don't know :) What do you think. I don't mind extracting it but it feels very risky in terms of BC as I expect breaking changes in this kind of modules way more often than React for instance.

It could be worth to bring up in a weekly JavaScript chat. Maybe (at least in the short-term) we can have some naming convention to designate some script handles as internal.

```php
wp_register_script( '_internal_react-spring', /* ... */ );
```

Reply:
I'm also curious whether there's anything which prohibits that a script handle be named where we embed the specific version. Obviously, unlike an "internal" designation, this might bound us to some compatibility so far as always shipping every version we've ever used, but at least it allows to migrate to newer versions over time.

```php
wp_register_script( 'react-spring@8.0.19', /* ... */ );
```

Reply:
This could work. We can define the versions for scripts in Core as well (argument of that function). Something in the PHP framework itself could work to register multiple versions and enqueue specific versions as well.

Reply:
> I'm also curious whether there's anything which prohibits that a script handle be named where we embed the specific version.

I remember now one of the issues we had considered with this is: When multiple versions are loaded in the same page, how do we handle globals being assigned without one version clobbering the other? I recall having some thoughts on using closure scoping and/or [accessor properties](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/defineProperty) to let the global vary depending on the consumer, but I hadn't ever formalized a proposal.